libcaliptra/src/caliptra_api.c

```diff
@@ -0,0 +1,277 @@
+// Licensed under the Apache-2.0 license
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <string.h>
+#include <errno.h>
+#include <caliptra_top_reg.h>
+#include "caliptra_if.h"
+#include "caliptra_api.h"
+#include "caliptra_fuses.h"
+#include "caliptra_mbox.h"
+
+#define CALIPTRA_STATUS_NOT_READY 0
+
+static inline uint32_t caliptra_read_status(void)
+{
+    uint32_t status;
+
+    caliptra_read_u32(CALIPTRA_TOP_REG_GENERIC_AND_FUSE_REG_CPTRA_FLOW_STATUS, &status);
+   
+    return status;
+}
+
+/**
+ * caliptra_ready_for_fuses
+ *
+ * Reports if the Caliptra hardware is ready for fuse data
+ *
+ * @return bool True if ready, false otherwise
+ */
+bool caliptra_ready_for_fuses(void)
+{
+    uint32_t status;
+    bool     ready = false;
+
+    status = caliptra_read_status();
+
+    if ((status & GENERIC_AND_FUSE_REG_CPTRA_FLOW_STATUS_READY_FOR_FUSES_MASK)
+                == GENERIC_AND_FUSE_REG_CPTRA_FLOW_STATUS_READY_FOR_FUSES_MASK)
+    {
+        return true;
+    }
```

Comment:
can this be done inline ? 

`return` status & GENERIC_AND_FUSE_REG_CPTRA_FLOW_STATUS_READY_FOR_FUSES_MASK)
                == GENERIC_AND_FUSE_REG_CPTRA_FLOW_STATUS_READY_FOR_FUSES_MASK);

Reply:
I had it verbose for readability but we can do it inline if desired.

Reply:
leave to others, but I think it can everybody should be able to understand the intent without being fully explicit...but leave this up to you

Reply:
+1 for making it inline. Also suggest making it as: return  ((caliptra_read_status() & CALIPTRA_TOP_REG_GENERIC_AND_FUSE_REG_CPTRA_FLOW_STATUS)
                == GENERIC_AND_FUSE_REG_CPTRA_FLOW_STATUS_READY_FOR_FUSES_MASK)